module: Added coverity module for static analysis

leemgs opened the pull request:
Fixed issue nnsuite/nnstreamer#1682.

This module is to examine the static analysis of C/C++ source code and reports potential issues.

Coverity is a static code analysis tool from Synopsys. This product enables engineers and security teams
to quickly find and fix defects and security vulnerabilities in custom source code written in C, C++,
Java, C#, JavaScript and more.

Coverity Scan is a free static-analysis cloud-based service for the open source community.
The tool analyzes over 3900 open-source projects and is integrated with GitHub and Travis CI.

**Reference**
 * Wiki page for this module: https://github.com/nnsuite/TAOS-CI/wiki/TAOS-CI-Plug-in-Modules
 * Module guide: https://github.com/nnsuite/TAOS-CI/blob/master/ci/doc/how-to-use-taos-ci-module.md

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

leemgs commented:
**Changelog**
* Version 4:
  - Added commands to configure the compiler type and compiler command.
  - Added check function if ccache exists

* Version 3:
  - Added an interconnection procedure between local-coverity and cloud-coverity (scan.coverity.com)
  - Added coverity-based build commands

* Version 2:
  - Added configuration variables.
  - Added document.
  - Added installation step of "coverity" package.

* Version 1:
  - Added new module (e.g. coverity) for static analysis.

leemgs commented:
* After this PR, if the coverity module submit the generated raw files to the coverity cloud (e.g., scan.coverity.com), we can see the below message (e.g.,  "Last Build Status: Running. Your build is currently being analyzed").
![image](https://user-images.githubusercontent.com/82404/67259979-cad8e680-f4d3-11e9-9be8-1429235e8449.png)

myungjoo commented on ci/taos/plugins-good/pr-prebuild-coverity.sh:
```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+##
+# Copyright (c) 2018 Samsung Electronics Co., Ltd. All Rights Reserved.
```
2019

leemgs replied:
Thanks. Fixed.

leemgs commented:
**Changelog**
    * Version 5:
      - Improved a readability of the scripts for maintenance.
      - Fixed the issue that cov-build does not make the result file.

    * Version 4:
      - Added commands to configure the compiler type and compiler command.
      - Added check function if ccache exists

    * Version 3:
      - Added an interconnection procedure between local-coverity and cloud-coverity (scan.coverity.com)
      - Added coverity-based build commands

    * Version 2:
      - Added configuration variables.
      - Added document.
      - Added installation step of "coverity" package.

    * Version 1:
      - Added new module (e.g. coverity) for static analysis.

dongju-chae commented:
One question here.
This script only checks the success of curl command (i.e., Coverity submission).
Then, how a PR submitter knows his PR has some defects or not?
So.. in auto-generated github comments or visiting coverity site?

leemgs commented:
> Then, how a PR submitter knows his PR has some defects or not?
> So.. in auto-generated github comments or visiting coverity site?

Goo Question. ^^ I am going to implement a web-crawler as I explained at the weekly meeting. 
The crawaler is responsible for  fetching required data from the Coverity cloud (e.g., scan.coverity.com)

leemgs commented:
## work status on Sep-17-2019 09:20

**Changelog**
* Version 6:
  - Added the build type of the coverity module
  - Re-arranged the execution procedure of the coverity module (totally 4 steps)
  - Added a check routine to know the status of the scan.coverity.com/projects/<github-prj-name>
    after sending a commit task with the curl command

* Version 5:
  - Improved a readability of the scripts for maintenance.
  - Fixed the issue that cov-build does not make the result file.
  - Renamed variables that are used  by cov-build command for configuration structure

* Version 4:
  - Added commands to configure the compiler type and compiler command.
  - Added check function if ccache exists

* Version 3:
  - Added an interconnection procedure between local-coverity and cloud-coverity (scan.coverity.com)
  - Added coverity-based build commands

* Version 2:
  - Added configuration variables.
  - Added document.
  - Added installation step of "coverity" package.

* Version 1:
  - Added new module (e.g. coverity) for static analysis.

leemgs commented:
To Reviewers, 

**Changelog**
* Version 7:
  - Added standalone coverity scan script for cron daemon
  - Added the web-crawling function
  - Added curl command for using the coverity cloud (e.g., scan.coverity.com)
  - Added configuration variables for the coverity module

* Version 6:
  - Added the build type of the coverity module
  - Re-arranged the execution procedure of the coverity module (totally 4 steps)
  - Added a check routine to know the status of the scan.coverity.com/projects/<github-prj-name>
    after sending a commit task with the curl command

* Version 5:
  - Improved a readability of the scripts for maintenance.
  - Fixed the issue that cov-build does not make the result file.
  - Renamed variables that are used  by cov-build command for configuration structure

* Version 4:
  - Added commands to configure the compiler type and compiler command.
  - Added check function if ccache exists

* Version 3:
  - Added an interconnection procedure between local-coverity and cloud-coverity (scan.coverity.com)
  - Added coverity-based build commands

* Version 2:
  - Added configuration variables.
  - Added document.
  - Added installation step of "coverity" package.

* Version 1:
  - Added new module (e.g. coverity) for static analysis.

leemgs commented:
This PR is ready to be merged because the local test is completely done.
